hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -216,6 +221,11 @@ public void rollbackMigration(PartitionMigrationEvent event) {
             removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(),
                     event.getCurrentReplicaIndex());
             getMetaDataGenerator().removeUuidAndSequence(event.getPartitionId());
+
+            if (event.getNewReplicaIndex() == -1) {
```

Comment:
should be `currentReplicaIndex` when rollback.

Reply:
Good catch, using current here.